lib/index.js

```diff
@@ -83,6 +85,19 @@ exports.request = function (method, url, options, callback, _trace) {
             req.abort();
         }
 
+        var logData = { err: err, url: url, options: options, req: req, beginTime: new Date().getTime() };
+
+        if (err) {
+            logger.log(logData);
+        }
+        else {
+            exports.read(res, null, function (err, payload) {
```

Comment:
this is going to be an expensive operation... this should only happen when we have logging enabled.  Please only do this part in the log function.